test/unit/org/apache/cassandra/batchlog/BatchlogEndpointFilterTest.java

```diff
@@ -56,7 +87,25 @@ public void shouldUseLocalRackIfPreferred() throws UnknownHostException
                                                                           .put("2", InetAddressAndPort.getByName("2"))
                                                                           .put("2", InetAddressAndPort.getByName("22"))
                                                                           .build();
-        Collection<InetAddressAndPort> result = filterBatchlogEndpoints(true, endpoints);
+        Collection<InetAddressAndPort> result = filterBatchlogEndpointsRandom(true, endpoints);
+        assertThat(result.size()).isEqualTo(2);
+        assertThat(result).containsAnyElementsOf(endpoints.get(LOCAL));
+        assertThat(result).containsAnyElementsOf(Iterables.concat(endpoints.get("1"), endpoints.get("2")));
+    }
+
+    @Test
+    public void shouldUseLocalRackIfPreferredStrategy() throws UnknownHostException
+    {
+        DatabaseDescriptor.setBatchlogEndpointStrategy(Config.BatchlogEndpointStrategy.prefer_local);
+        Multimap<String, InetAddressAndPort> endpoints = ImmutableMultimap.<String, InetAddressAndPort> builder()
+                .put(LOCAL, InetAddressAndPort.getByName("0"))
+                .put(LOCAL, InetAddressAndPort.getByName("00"))
+                .put("1", InetAddressAndPort.getByName("1"))
+                .put("1", InetAddressAndPort.getByName("11"))
+                .put("2", InetAddressAndPort.getByName("2"))
+                .put("2", InetAddressAndPort.getByName("22"))
+                .build();
+        Collection<InetAddressAndPort> result = filterBatchlogEndpointsRandom(false, endpoints);
```

Comment:
based on the definition of `filterBatchlogEndpointsRandom` the first argument says if the local is preferred. How come you set it to false in this test ?